File: past_archive/boj8/2851.cc
// 슈퍼 마리오
// https://www.acmicpc.net/problem/2851

#include <stdio.h>

int Abs(int x) { return x < 0 ? -x : x; }

int main() {
  int sum = 0;
  for (int i = 0; i < 10; ++i) {
    int x;
    scanf("%d", &x);
    int tmp_sum = sum + x;
    int diff = Abs(tmp_sum - 100);
    if (diff < Abs(sum - 100)) {
      sum = tmp_sum;
    } else if (diff == Abs(sum - 100)) {
      if (tmp_sum > sum) {
        sum = tmp_sum;
      }
    } else {
      break;
    }
  }
  printf("%d\n", sum);
  return 0;
}

File: 24boj_py/2851.py
# 슈퍼 마리오
# https://www.acmicpc.net/problem/2851

import sys

# read until EOF, integer will be given line by line
# and store them in a list
mushrooms = [int(sys.stdin.readline()) for _ in range(10)]

s = 0
for mushroom in mushrooms:
    tmp = s + mushroom
    if abs(tmp - 100) <= abs(s - 100):
        s = tmp
    else:
        break

sys.stdout.write(f'{s}\n')